### PR TITLE
More broadly used GBPTreeBuilder for tests

### PR DESCRIPTION
**Pull request by tinwelint**

### CODE REVIEW

**Review by klaren — CHANGES_REQUESTED** [review]
Some tiny details

File: community/index/src/test/java/org/neo4j/index/internal/gbptree/GBPTreeBuilder.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+class GBPTreeBuilder<KEY,VALUE>
+{
+    PageCache pageCache;
+    File file;
+    int tentativePageSize = 0;
+    Monitor monitor = NO_MONITOR;
+    Header.Reader headerReader = NO_HEADER_READER;
+    Layout<KEY,VALUE> layout;
+    Consumer<PageCursor> headerWriter = NO_HEADER_WRITER;
+
```
Comment (klaren) [inline community/index/src/test/java/org/neo4j/index/internal/gbptree/GBPTreeBuilder.java]: Fields should be private for a cleaner builder

File: community/index/src/test/java/org/neo4j/index/internal/gbptree/FormatCompatibilityTest.java [hunk 2]
```diff
@@ -97,8 +94,8 @@ public void shouldDetectFormatChange() throws Throwable
         // WHEN reading from the tree
         // THEN everything should work, otherwise there has likely been a format change
         PageCache pageCache = pageCacheRule.getPageCache( fsRule.get() );
-        try ( GBPTree<MutableLong,MutableLong> tree = new GBPTree<>( pageCache, storeFile, new SimpleLongLayout(), 0,
-                NO_MONITOR, NO_HEADER_READER, NO_HEADER_WRITER ) )
+        try ( GBPTree<MutableLong,MutableLong> tree =
+                new GBPTreeBuilder( pageCache, storeFile, new SimpleLongLayout() ).build() )
```
Comment (klaren) [inline community/index/src/test/java/org/neo4j/index/internal/gbptree/FormatCompatibilityTest.java]: `GBPTreeBuilder( ... )` is unchecked. Should be `GBPTreeBuilder<>( ... )`. My search turned up with 13 occurrences of this, the same comment applies to all of them.

**Comment by tinwelint** [comment]
@klaren comments addressed

**Review by klaren — APPROVED** [review]
Looks good!

**Pull request merged**